script/ui-controller.js

```diff
@@ -47,12 +50,55 @@ class UIController {
 		this.errorMessage.classList.add('hidden');
 	}
 
+	showToast(message) {
+		this.toastMessage.textContent = message;
+		this.toastMessage.classList.remove('hidden', 'error');
+
+		setTimeout(() => this.toastMessage.classList.add('shown'), 10);
+		setTimeout(() => this.hideToast(), 3000);
+	}
+
+	showErrorToast(message) {
+		this.toastMessage.textContent = message;
+		this.toastMessage.classList.remove('hidden');
+		this.toastMessage.classList.add('error');
+
+		setTimeout(() => this.toastMessage.classList.add('shown'), 10);
+		setTimeout(() => this.hideToast(), 3000);
+	}
+
+	hideToast() {
+		this.toastMessage.classList.remove('shown');
+
+		setTimeout(
+			() => {
+				this.toastMessage.classList.add('hidden');
+				this.toastMessage.classList.remove('error');
+			},
+			300
+		);
+	}
+
 	/**
 	 * @param {WallpaperResponse} wallpaperResponse
 	 */
 	updateWallpaperInfo({ id, url, resolution, category, purity }) {
 		this.infoText.textContent = `${id} | ${resolution} | ${category} | ${purity}`;
-		this.wallhavenLink.href = url;
+
+		if (this.onClickLink !== null) {
+			this.wallhavenLink.removeEventListener('click', this.onClickLink);
+		}
+
+		this.onClickLink = async () => {
+			try {
```

Comment:
<img width="563" height="188" alt="Image" src="https://github.com/user-attachments/assets/247711cd-b112-4700-8af0-d2491d7175d5" />

[The linter](https://app.codacy.com/gh/Attacktive/wallhavend-html/pull-requests/18?pr=18) has absolutely lost their mind so I'm not gonna do anything about it. 🙅